### PR TITLE
Create Core-Infrastructure and Datarepo-app modules

### PR DESCRIPTION
**Pull request by smark88**
Create reusable modules for deploying a DR for Terra and other.

### CODE REVIEW

**Comment by smark88** [comment]
Please review.

File: environments/dev-developer/main.tf [hunk 1]
```diff
@@ -0,0 +1,61 @@
+# gcp networking, k8 cluster
+module "core-infrastructure" {
+  source = "github.com/broadinstitute/terraform-jade.git//modules/core-infrastructure?ref=ms-datarepomodule"
```
Comment (thathert) [inline environments/dev-developer/main.tf:3]: Do you still need the refs here?

File: environments/dev-developer/main.tf [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+  providers = {
+    google.target      = google
+    google-beta.target = google-beta
+  }
+}
+
+# dns ips, sql server and dbs
+module "datarepo-app" {
+  source = "github.com/broadinstitute/terraform-jade.git//modules/datarepo-app?ref=ms-datarepomodule"
```
Comment (thathert) [inline environments/dev-developer/main.tf:21]: and here?

File: modules/core-infrastructure/network.tf [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+    {
+      range_name    = "pods"
+      ip_cidr_range = var.gke_subnet_pods
+    },
+    {
+      range_name    = "services"
+      ip_cidr_range = var.gke_subnet_services
+  }]
+  dynamic "log_config" {
+    for_each = var.enable_flow_logs ? ["If only TF supported if/else"] : []
```
Comment (thathert) [inline modules/core-infrastructure/network.tf:36]: I was about to comment about why this isn't an if/else and then read the string..

File: modules/core-infrastructure/nat.tf [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+  network    = google_compute_network.network[0].self_link
+  depends_on = [google_compute_network.network, google_compute_subnetwork.subnetwork, var.dependencies]
+
+  bgp {
+    asn = 64514
+  }
+}
+
+resource "google_compute_address" "nat-address" {
+  count      = var.enable ? 2 : 0
```
Comment (thathert) [inline modules/core-infrastructure/nat.tf:17]: why two nat addresses?

File: modules/core-infrastructure/nat.tf [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+resource "google_compute_router" "router" {
+  count      = var.enable ? 1 : 0
+  provider   = google.target
+  name       = "router"
+  project    = var.google_project
+  network    = google_compute_network.network[0].self_link
+  depends_on = [google_compute_network.network, google_compute_subnetwork.subnetwork, var.dependencies]
+
+  bgp {
+    asn = 64514
```
Comment (thathert) [inline modules/core-infrastructure/nat.tf:12]: what's this do? comment would be helpful

File: modules/core-infrastructure/provider.tf [hunk 6]
```diff
@@ -0,0 +1,7 @@
+provider "google" {
+  alias = "target"
+}
+
+provider "google-beta" {
+  alias = "target"
```
Comment (thathert) [inline modules/core-infrastructure/provider.tf:6]: feels weird to have them both share the same alias, what's going on here?

File: modules/datarepo-app/variables.tf [hunk 7]
```diff
@@ -0,0 +1,160 @@
...
+    "roles/bigquery.admin",
+    "roles/cloudsql.admin",
+    "roles/datastore.user",
+    "roles/errorreporting.writer",
+    "roles/logging.logWriter",
+    "roles/monitoring.admin",
+    "roles/servicemanagement.serviceController",
+    "roles/stackdriver.accounts.viewer",
+    "roles/storage.admin",
+    "roles/pubsub.admin"
```
Comment (thathert) [inline modules/datarepo-app/variables.tf:54]: what about kubernetes admin? Is this complete?

File: environments/dev-developer/main.tf (hunk 2 above)
Comment (smark88) [reply environments/dev-developer/main.tf:21]: see above

File: modules/core-infrastructure/nat.tf (hunk 4 above)
Comment (smark88) [reply modules/core-infrastructure/nat.tf:17]: sure we always have.

File: modules/core-infrastructure/nat.tf (hunk 5 above)
Comment (smark88) [reply modules/core-infrastructure/nat.tf:12]: BGP Autonomous System Number for router

File: modules/core-infrastructure/provider.tf (hunk 6 above)
Comment (smark88) [reply modules/core-infrastructure/provider.tf:6]: it works out to this when used
 ```
  providers = {
    google.target      = google
    google-beta.target = google-beta
  }
```

Comment (smark88) [reply modules/core-infrastructure/provider.tf:6]: target is just used as a name for the pass through to the module

**Pull request merged**